ru/lessons/advanced/umbrella-projects.md

```diff
@@ -0,0 +1,212 @@
+* creating test/test_helper.exs
+* creating test/utilities_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd utilities
+    mix test
+
+Run "mix help" for more commands.
+
+
+$ mix new datasets
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/datasets.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/datasets_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd datasets
+    mix test
+
+Run "mix help" for more commands.
+
+$ mix new svm --sup
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/svm.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/svm_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd svm
+    mix test
+
+Run "mix help" for more commands.
+```
+
+В результате структура проекта должна получиться такой:
+
+```shell
+$ tree
+.
+├── README.md
+├── apps
+│   ├── datasets
+│   │   ├── README.md
+│   │   ├── config
+│   │   │   └── config.exs
+│   │   ├── lib
+│   │   │   └── datasets.ex
+│   │   ├── mix.exs
+│   │   └── test
+│   │       ├── datasets_test.exs
+│   │       └── test_helper.exs
+│   ├── svm
+│   │   ├── README.md
+│   │   ├── config
+│   │   │   └── config.exs
+│   │   ├── lib
+│   │   │   └── svm.ex
+│   │   ├── mix.exs
+│   │   └── test
+│   │       ├── svm_test.exs
+│   │       └── test_helper.exs
+│   └── utilities
+│       ├── README.md
+│       ├── config
+│       │   └── config.exs
+│       ├── lib
+│       │   └── utilities.ex
+│       ├── mix.exs
+│       └── test
+│           ├── test_helper.exs
+│           └── utilities_test.exs
+├── config
+│   └── config.exs
+└── mix.exs
+```
+
+Если мы перейдём обратно, к корню зонтичного проекта, то окажется, что мы можем использовать все обычные команды `Mix`, такие как компиляция. Так как все под-проекты это обычные приложения, вы можете перейти в их директории, и выполнять все те действия, которые вам обычно позволяет `Mix`.
+
+```
+$ mix compile
+
+==> svm
+Compiled lib/svm.ex
+Generated svm app
+
+==> datasets
+Compiled lib/datasets.ex
+Generated datasets app
+
+==> utilities
+Compiled lib/utilities.ex
+Generated utilities app
+
+Consolidated List.Chars
+Consolidated Collectable
+Consolidated String.Chars
+Consolidated Enumerable
+Consolidated IEx.Info
+Consolidated Inspect
+```
+
+## IEx
+
+Вы можете подумать, что взаимодействие с приложениями в зонтичном проекте будет немного отличаться? На самом деле это не так! Если мы перейдём в корневую директорию проекта и запустим `IEx` командой `iex -S mix` &mdash; мы сможем нормально взаимодействовать со всеми проектами. Давайте, для примера, изменим код приложения `apps/datasets/lib/datasets.ex`:
```

Comment:
I suppose that there should be no question mark at the end of the first sentence.